### PR TITLE
added cinic10-dataset

### PR DESCRIPTION
**Pull request by da-h**
Hi,

this PR adds the [**CINIC-10**](https://www.bayeswatch.com/2018/10/09/CINIC) to *tf-boilerplate*.

As the other scripts, this script
- downloads the tar-file
- extracts all contents
- converts the png-files to tfrecords

Best
*da-h*

### CODE REVIEW

File: data/cinic10/__init__.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+HEIGHT = 32
+WIDTH = 32
+DEPTH = 3
+datashape = [HEIGHT,WIDTH,DEPTH]
+numexamples = 270000
+cinic_mean = [0.47889522, 0.47227842, 0.43047404]
+cinic_std = [0.24205776, 0.23828046, 0.25874835]
+
+
+class Cifar10DataSet(object):
```
Comment (sbrodehl) [inline data/cinic10/__init__.py]: The class name needs to be changes.
```suggestion
class Cinic10DataSet(object):
```

File: data/cinic10/__init__.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
+"""CINIC-10 data set.
+
+See https://datashare.is.ed.ac.uk/handle/10283/3192
+"""
+import os
+import tensorflow as tf
+from template import BaseDataSampler
+from .generate_cifar10_tfrecords import main as download
```
Comment (sbrodehl) [inline data/cinic10/__init__.py]: is `generate_cifar10_tfrecords` correct?

File: data/cinic10/__init__.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        filepath = os.path.join(self.data_dir, "train.tfrecords")
+        if not tf.gfile.Exists(self.data_dir):
+            tf.gfile.MakeDirs(self.data_dir)
+        if not tf.gfile.Exists(filepath):
+            print("Downloading to "+filepath)
+            print("Please wait...")
+            download(self.data_dir)
+
+    def training(self):
+        return Cifar10DataSet(self.data_dir, subset='train', use_distortion=False).make_batch()
```
Comment (sbrodehl) [inline data/cinic10/__init__.py]: Needs to be changed as well
```suggestion
        return Cinic10DataSet(self.data_dir, subset='train', use_distortion=False).make_batch()
```

File: data/cinic10/__init__.py [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+    def testing(self):
+        return Cifar10DataSet(self.data_dir, subset='eval', use_distortion=False).make_batch()
```
Comment (sbrodehl) [inline data/cinic10/__init__.py]: Needs to be changed as well
```suggestion
        return Cinic10DataSet(self.data_dir, subset='eval', use_distortion=False).make_batch()
```

**Review by sbrodehl — CHANGES_REQUESTED** [review]
Thanks for the contribution!
The cinic-10 dataset would be a great addition to the project.

Please review the suggested changes.

**Comment by da-h** [comment]
Hi,

I cleaned up the commit, adding the changes that you suggested.

Best
*da-h*

**Comment by sbrodehl** [comment]
Thank you!

**Pull request merged**